### PR TITLE
Removed variables from canvas

### PR DESCRIPTION
**Pull request by davidpiuva**
Some variables are deleted and some things are moved to the OpenGL layer.
Builds on https://github.com/magic-lang/ooc-kean/pull/1637

Waiting to rebase on upstream before peer review can start.

### CODE REVIEW

**Comment by marcusnaslund** [comment]
#1637 now merged

**Comment by davidpiuva** [comment]
Rebased

File: source/draw/gpu/opengl/OpenGLSurface.ooc [hunk 1]
```diff
@@ -23,17 +24,48 @@ OpenGLSurface: abstract class extends GpuCanvas {
 	_unbind: abstract func
 	drawLines: override func ~explicit (pointList: VectorList<FloatPoint2D>, pen: Pen) {
 		this _bind()
-		this context backend setViewport(this viewport)
+		this context backend setViewport(IntBox2D new(this size))
 		this context backend enableBlend(false)
-		this context drawLines(pointList, this _projection * this _toLocal, pen)
+		this context drawLines(pointList, this _createProjection(this size toFloatVector2D(), 0.0f) * this _toLocal, pen)
```
Comment (davidpiuva) [inline source/draw/gpu/opengl/OpenGLSurface.ooc]: Creating an isometric projection matrix for 2D drawing in reference coordinates.

File: source/draw/gpu/opengl/OpenGLSurface.ooc [hunk 2]
```diff
@@ -23,17 +24,48 @@ OpenGLSurface: abstract class extends GpuCanvas {
 	_unbind: abstract func
 	drawLines: override func ~explicit (pointList: VectorList<FloatPoint2D>, pen: Pen) {
 		this _bind()
-		this context backend setViewport(this viewport)
+		this context backend setViewport(IntBox2D new(this size))
```
Comment (davidpiuva) [inline source/draw/gpu/opengl/OpenGLSurface.ooc]: Using a custom viewport for line drawing is overkill for debugging purposes.

File: source/draw/gpu/opengl/OpenGLSurface.ooc [hunk 3]
```diff
@@ -15,6 +15,7 @@ import backend/GLRenderer
 
 version(!gpuOff) {
 OpenGLSurface: abstract class extends GpuCanvas {
+	_toLocal := FloatTransform3D createScaling(1.0f, -1.0f, -1.0f)
```
Comment (davidpiuva) [inline source/draw/gpu/opengl/OpenGLSurface.ooc]: Moved from GpuCanvas because it did not belong there.

File: source/draw/gpu/GpuCanvas.ooc [hunk 4]
```diff
@@ -16,50 +16,9 @@ import GpuContext, GpuImage, Mesh, GpuYuv420Semiplanar
 version(!gpuOff) {
 GpuCanvas: abstract class extends Canvas {
 	_context: GpuContext
-	_model: FloatTransform3D // Deprecated! Do not use.
-	_view := FloatTransform3D identity // Deprecated! Do not use.
-	_projection: FloatTransform3D // Deprecated! Do not use.
-	_toLocal := FloatTransform3D createScaling(1.0f, -1.0f, -1.0f)
```
Comment (davidpiuva) [inline source/draw/gpu/GpuCanvas.ooc]: `_toLocal` was moved to OpenGLSurface.

File: source/draw/gpu/opengl/OpenGLSurface.ooc [hunk 5]
```diff
@@ -23,17 +24,48 @@ OpenGLSurface: abstract class extends GpuCanvas {
...
+			farPlane := 10000.0f
+			a := flipX * 2.0f * focalLengthPerWidth
+			f := flipY * 2.0f * focalLengthPerHeight
+			k := (farPlane + nearPlane) / (farPlane - nearPlane)
+			o := 2.0f * farPlane * nearPlane / (farPlane - nearPlane)
+			result = FloatTransform3D new(a, 0.0f, 0.0f, 0.0f, 0.0f, f, 0.0f, 0.0f, 0.0f, 0.0f, k, -1.0f, 0.0f, 0.0f, o, 0.0f)
+		} else
+			result = FloatTransform3D createScaling(flipX * 2.0f / targetSize x, flipY * 2.0f / targetSize y, 0.0f)
+		result
+	}
```
Comment (davidpiuva) [inline source/draw/gpu/opengl/OpenGLSurface.ooc]: All create methods are moved from GpuCanvas because they are specific to OpenGL's coordinate systems.

**Comment by davidpiuva** [comment]
I will let @emilwestergren review it since he should be back tomorrow.

File: source/draw/gpu/GpuCanvasYuv420Semiplanar.ooc [hunk 6]
```diff
@@ -15,22 +15,6 @@ import GpuImage, Map, GpuCanvas, GpuContext, GpuYuv420Semiplanar, Mesh
...
-		}
-	}
-	focalLength: Float {
-		get { this _focalLength }
-		set(value) {
-			this _focalLength = value
-			this _target y canvas focalLength = value
-			this _target uv canvas focalLength = value / 2
-		}
-	}
```
Comment (davidpiuva) [inline source/draw/gpu/GpuCanvasYuv420Semiplanar.ooc]: Removed.

File: source/draw/gpu/GpuCanvas.ooc [hunk 7]
```diff
@@ -16,50 +16,9 @@ import GpuContext, GpuImage, Mesh, GpuYuv420Semiplanar
...
-				f := -(this _coordinateTransform e as Float) * 2.0f * this _focalLength / this size y
-				k := (farPlane + nearPlane) / (farPlane - nearPlane)
-				o := 2.0f * farPlane * nearPlane / (farPlane - nearPlane)
-				this _projection = FloatTransform3D new(a, 0.0f, 0.0f, 0.0f, 0.0f, f, 0.0f, 0.0f, 0.0f, 0.0f, k, -1.0f, 0.0f, 0.0f, o, 0.0f)
-			}
-			else
-				this _projection = FloatTransform3D createScaling(2.0f / this size x, -(this _coordinateTransform e as Float) * 2.0f / this size y, 1.0f)
-			this _model = this _createModelTransform(IntBox2D new(this size), this _focalLength)
-		}
-	}
```
Comment (davidpiuva) [inline source/draw/gpu/GpuCanvas.ooc]: Removed

File: source/draw/gpu/GpuCanvas.ooc [hunk 8]
```diff
@@ -16,50 +16,9 @@ import GpuContext, GpuImage, Mesh, GpuYuv420Semiplanar
...
-		this _createModelTransform(box toFloatBox2D(), focalLength)
-	}
-	_createModelTransform: func ~LocalFloat (box: FloatBox2D, focalLength: Float) -> FloatTransform3D {
-		toReference := FloatTransform3D createTranslation((box size x - this size x) / 2, (this size y - box size y) / 2, 0.0f)
-		translation := this _toLocal * FloatTransform3D createTranslation(box leftTop x, box leftTop y, focalLength) * this _toLocal
-		translation * toReference * FloatTransform3D createScaling(box size x / 2.0f, box size y / 2.0f, 1.0f)
-	}
-	_createModelTransformNormalized: func (imageSize: IntVector2D, box: FloatBox2D, focalLength: Float) -> FloatTransform3D {
-		this _createModelTransform(box * imageSize toFloatVector2D(), focalLength)
-	}
```
Comment (davidpiuva) [inline source/draw/gpu/GpuCanvas.ooc]: Moved to OpenGLSurface.

File: source/draw/gpu/GpuCanvas.ooc [hunk 9]
```diff
@@ -72,25 +31,5 @@ GpuCanvas: abstract class extends Canvas {
...
-			farPlane := 10000.0f
-			a := flipX * 2.0f * focalLengthPerWidth
-			f := flipY * 2.0f * focalLengthPerHeight
-			k := (farPlane + nearPlane) / (farPlane - nearPlane)
-			o := 2.0f * farPlane * nearPlane / (farPlane - nearPlane)
-			result = FloatTransform3D new(a, 0.0f, 0.0f, 0.0f, 0.0f, f, 0.0f, 0.0f, 0.0f, 0.0f, k, -1.0f, 0.0f, 0.0f, o, 0.0f)
-		} else
-			result = FloatTransform3D createScaling(flipX * 2.0f / targetSize x, flipY * 2.0f / targetSize y, 0.0f)
-		result
-	}
```
Comment (davidpiuva) [inline source/draw/gpu/GpuCanvas.ooc]: Moved to OpenGLSurface.

File: source/draw/gpu/GpuCanvas.ooc [hunk 10]
```diff
@@ -16,50 +16,9 @@ import GpuContext, GpuImage, Mesh, GpuYuv420Semiplanar
 version(!gpuOff) {
 GpuCanvas: abstract class extends Canvas {
 	_context: GpuContext
-	_model: FloatTransform3D // Deprecated! Do not use.
-	_view := FloatTransform3D identity // Deprecated! Do not use.
-	_projection: FloatTransform3D // Deprecated! Do not use.
-	_toLocal := FloatTransform3D createScaling(1.0f, -1.0f, -1.0f)
-	_focalLength: Float // Deprecated! Do not use.
```
Comment (davidpiuva) [inline source/draw/gpu/GpuCanvas.ooc]: `DrawState` already has Focal length.

File: source/draw/gpu/GpuCanvas.ooc [hunk 11]
```diff
@@ -16,50 +16,9 @@ import GpuContext, GpuImage, Mesh, GpuYuv420Semiplanar
 version(!gpuOff) {
 GpuCanvas: abstract class extends Canvas {
 	_context: GpuContext
-	_model: FloatTransform3D // Deprecated! Do not use.
-	_view := FloatTransform3D identity // Deprecated! Do not use.
-	_projection: FloatTransform3D // Deprecated! Do not use.
```
Comment (davidpiuva) [inline source/draw/gpu/GpuCanvas.ooc]: `DrawState` contains `view`, generates `model` from `destination` and generates `projection` from `focalLength`

File: source/draw/gpu/GpuCanvas.ooc [hunk 12]
```diff
@@ -16,50 +16,9 @@ import GpuContext, GpuImage, Mesh, GpuYuv420Semiplanar
...
-			}
-			else
-				this _projection = FloatTransform3D createScaling(2.0f / this size x, -(this _coordinateTransform e as Float) * 2.0f / this size y, 1.0f)
-			this _model = this _createModelTransform(IntBox2D new(this size), this _focalLength)
-		}
-	}
 	init: func (size: IntVector2D, =_context, =_defaultMap, =_coordinateTransform) { super(size) }
-	_createModelTransform: func ~LocalInt (box: IntBox2D, focalLength: Float) -> FloatTransform3D {
-		this _createModelTransform(box toFloatBox2D(), focalLength)
-	}
```
Comment (davidpiuva) [inline source/draw/gpu/GpuCanvas.ooc]: This integer wrapper was no longer used.

File: source/draw/Canvas.ooc [hunk 13]
```diff
@@ -21,19 +21,13 @@ InterpolationMode: enum {
 
 Canvas: abstract class {
 	_size: IntVector2D
-	_transform := FloatTransform3D identity
 	size ::= this _size
 	viewport: IntBox2D { get set }
-	blend: Bool { get set }
-	opacity: Float { get set }
-	transform: FloatTransform3D { get { this _transform } set(value) { this _transform = value } }
```
Comment (davidpiuva) [inline source/draw/Canvas.ooc]: The transform was not even used by the legacy draw calls on the CPU. Just blit operations for resizing things.

**Comment by davidpiuva** [comment]
@marcusnaslund Ready for final review.

**Comment by marcusnaslund** [comment]
Looks amazing.

**Pull request merged**